src/cbor/decoder.rs

```diff
@@ -11,6 +11,10 @@ struct DecoderCursor {
 impl DecoderCursor {
     /// Read and return the given number of bytes from the cursor. Advances the cursor.
     fn read_bytes(&mut self, len: usize) -> Result<Vec<u8>, CborError> {
+        // We can not allocate more than 2GB.
+        if len > 2147483648 {
```

Comment:
Should we expose this as a constant or something?

Reply:
Yeah, why not.